src/Microsoft.AspNetCore.Authentication.OpenIdConnect/ClaimActionCollectionUniqueExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Security.Claims;
+using Microsoft.AspNetCore.Authentication.OpenIdConnect;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    public static class ClaimActionCollectionUniqueExtensions
+    {
+        /// <summary>
+        /// Selects a top level value from the json user data with the given key name and adds it as a Claim.
+        /// This no-ops if the ClaimsIdentity already contains a Claim with the given ClaimType.
+        /// This no-ops if the key is not found or the value is empty.
+        /// </summary>
+        /// <param name="collection"></param>
+        /// <param name="claimType">The value to use for Claim.Type when creating a Claim.</param>
+        /// <param name="jsonKey">The top level key to look for in the json user data.</param>
+        public static void MapUniqueJsonKey(this ClaimActionCollection collection, string claimType, string jsonKey)
+        {
+            collection.MapUniqueJsonKey(claimType, jsonKey, ClaimValueTypes.String);
+        }
+
+        /// <summary>
+        /// Selects a top level value from the json user data with the given key name and adds it as a Claim.
+        /// This no-ops if the ClaimsIdentity already contains a Claim with the given ClaimType.
+        /// This no-ops if the key is not found or the value is empty.
+        /// </summary>
+        /// <param name="collection"></param>
+        /// <param name="claimType">The value to use for Claim.Type when creating a Claim.</param>
+        /// <param name="jsonKey">The top level key to look for in the json user data.</param>
+        /// <param name="valueType">The value to use for Claim.ValueType when creating a Claim.</param>
+        public static void MapUniqueJsonKey(this ClaimActionCollection collection, string claimType, string jsonKey, string valueType)
+        {
+            collection.Add(new UniqueJsonKeyClaimAction(claimType, valueType, jsonKey));
```

Comment:
So my earlier point about the Collection vs Manager wasn't so much about that name, but that it seems unnecessary to have public types that are only used in the extension method.  Why can't we just bake the logic directly into Method instead of having the method and a type (that's just a class around a Action.  Cleans up the public surface area, and I don't see much use for the action types outside of the methods.  Which lead me to ask why we shouldn't just call it a Manager with the methods, instead of a List of action thingies...